<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>864</width>
    <height>494</height>
   </rect>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow
{
	border: 2px solid  rgb(100, 100, 100);
}

#draggableTitleBar
{
	background: rgb(70, 70, 70);
	border-radius: 0px;
	border: 2px solid  rgb(70, 70, 70);
}

QGroupBox
{
	font: bold 18px;
	border-radius: 0px;
}
QGroupBox::title
{
	padding-top: 5px;
	subcontrol-position: top center;
}

QPushButton
{
	font: bold 18px;
	border: 2px solid #8f8f8f;
	background-color: #f8f8f8;
	border-radius: 5px;
}
QPushButton:pressed
{
	background-color: #e2e2e2;
}
QPushButton:hover
{
	border: 2px solid #cc6600;
}

QLineEdit
{
	font: bold 18px;
	border: 2px solid #8f8f8f;
	background-color: #f8f8f8;
	border-radius: 5px;
}
QLineEdit:focus
{
	border: 2px solid #cc6600;
}
QLineEdit:hover
{
	border: 2px solid #cc6600;
}

QComboBox
{
	font: bold 18px;
	border: 2px solid #8f8f8f;
	background-color: #f8f8f8;
	border-radius: 5px;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 15px;

    border-left-width: 1px;
    border-left-color: darkgray;
    border-left-style: solid; /* just a single line */
    border-top-right-radius: 3px; /* same radius as the QComboBox */
    border-bottom-right-radius: 3px;
}
QComboBox::down-arrow
{
	image: url(:/files/images/arrow_down.png);
	height: 12px;
	width: 12px;
}
QComboBox::on
{
	border: 2px solid #cc6600;
}
QComboBox:hover
{
	border: 2px solid #cc6600;
}
QComboBox QAbstractItemView
{
	border: 2px solid #cc6600;
	selection-background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #111, stop: 1 #333);
}

QPlainTextEdit
{
	font: bold 18px;
	border: 2px solid #8f8f8f;
	background-color: #f8f8f8;
	border-radius: 5px;
}

QLabel
{
	font: bold 18px;
}
</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="grpControlMode">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>42</y>
      <width>260</width>
      <height>300</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox
{
	background: rgba(0, 0, 0, 20);
}</string>
    </property>
    <property name="title">
     <string>Control Mode</string>
    </property>
    <widget class="QPushButton" name="btnHome">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>130</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Home</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnPointing">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>130</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Pointing</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnTracking">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>190</y>
       <width>130</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Tracking</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnManual">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>130</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Manual</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnStop">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>240</y>
       <width>130</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnGetMode">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>40</y>
       <width>70</width>
       <height>240</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #33cc33;
	background-color: #33cc33;
}
QPushButton:pressed
{
	background-color: #2eb82e;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}</string>
     </property>
     <property name="text">
      <string>Get 
 Mode</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grpPosVel">
    <property name="geometry">
     <rect>
      <x>262</x>
      <y>212</y>
      <width>600</width>
      <height>130</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox
{
	background: rgba(0, 0, 0, 70);
}</string>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QLineEdit" name="ledPos">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>20</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ledVel">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>70</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnGetPos">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>20</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #33cc33;
	background-color: #33cc33;
}
QPushButton:pressed
{
	background-color: #2eb82e;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}</string>
     </property>
     <property name="text">
      <string>Get Pos</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetPos">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #e67300;
	background-color: #e67300;
}
QPushButton:pressed
{
	background-color: #cc6600;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}</string>
     </property>
     <property name="text">
      <string>Set Pos</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetVel">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>70</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #e67300;
	background-color: #e67300;
}
QPushButton:pressed
{
	background-color: #cc6600;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}</string>
     </property>
     <property name="text">
      <string>Set Vel</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblPosUnit">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>20</y>
       <width>60</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>deg</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblVelUnit">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>70</y>
       <width>60</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>deg/s</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblPos">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>90</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Position:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblVel">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>90</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Velocity:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnGetVel">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>70</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #33cc33;
	background-color: #33cc33;
}
QPushButton:pressed
{
	background-color: #2eb82e;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}
QPushButton:disabled 
{
	border: 2px solid #606060;
	background-color: #606060;
}</string>
     </property>
     <property name="text">
      <string>Get Vel</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetPosVel">
     <property name="geometry">
      <rect>
       <x>510</x>
       <y>20</y>
       <width>70</width>
       <height>90</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #e67300;
	background-color: #e67300;
}
QPushButton:pressed
{
	background-color: #cc6600;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}</string>
     </property>
     <property name="text">
      <string>Set
Both</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grp_PIDTunner">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>342</y>
      <width>860</width>
      <height>150</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox
{
	background: rgba(0, 0, 0, 95);
}</string>
    </property>
    <property name="title">
     <string>PID TUNNER</string>
    </property>
    <widget class="QLineEdit" name="ledKff1">
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>40</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetKi">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>90</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Ki</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetKd">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>90</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Kd</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetKp">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>90</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Kp</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="ledKp">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>40</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetParams">
     <property name="geometry">
      <rect>
       <x>740</x>
       <y>90</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #e67300;
	background-color: #e67300;
}
QPushButton:pressed
{
	background-color: #cc6600;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}</string>
     </property>
     <property name="text">
      <string>Set All</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="ledKi">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>40</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ledKd">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>40</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnGetParams">
     <property name="geometry">
      <rect>
       <x>740</x>
       <y>40</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #33cc33;
	background-color: #33cc33;
}
QPushButton:pressed
{
	background-color: #2eb82e;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}</string>
     </property>
     <property name="text">
      <string>Get All</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetKff2">
     <property name="geometry">
      <rect>
       <x>620</x>
       <y>90</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Kff2</string>
     </property>
    </widget>
    <widget class="QComboBox" name="cboAxis">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="currentText">
      <string/>
     </property>
    </widget>
    <widget class="QComboBox" name="cboPIDName">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="ledKff2">
     <property name="geometry">
      <rect>
       <x>620</x>
       <y>40</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSetKff1">
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>90</y>
       <width>100</width>
       <height>40</height>
      </rect>
     </property>
     <property name="text">
      <string>Kff1</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grpComUart">
    <property name="geometry">
     <rect>
      <x>262</x>
      <y>42</y>
      <width>600</width>
      <height>170</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox
{
	background: rgba(0, 0, 0, 45);
}</string>
    </property>
    <property name="title">
     <string>COM UART</string>
    </property>
    <widget class="QPushButton" name="btnConnect">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>170</width>
       <height>60</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton
{
	font: bold 18px;
	color: white;
	border: 2px solid #33cc33;
	background-color: #33cc33;
}
QPushButton:pressed
{
	background-color: #2eb82e;
}
QPushButton:hover
{
	border: 2px solid #8f8f8f;
}</string>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
    <widget class="QComboBox" name="cboCom">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>170</width>
       <height>40</height>
      </rect>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="pteStatus">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>40</y>
       <width>380</width>
       <height>110</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="DraggableWidget" name="draggableTitleBar" native="true">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>2</y>
      <width>780</width>
      <height>40</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox
{
	background: rgb(70, 70, 70);
	border-radius: 0px;
	border: 2px solid  rgb(70, 70, 70);
}</string>
    </property>
    <widget class="QPushButton" name="btnIcon">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>40</width>
       <height>40</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton:disabled 
{
	border: none;
	background-color: none;
}</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset>
       <disabledoff>:/files/images/icon.png</disabledoff>
       <disabledon>:/files/images/icon.png</disabledon>
      </iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>35</width>
       <height>35</height>
      </size>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="btnClose">
    <property name="geometry">
     <rect>
      <x>822</x>
      <y>2</y>
      <width>40</width>
      <height>40</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton
{
	border-radius: 0px;
	border: none;
	background-color: #ff0000;
}
QPushButton:pressed
{
	background-color: #ff0000;
}
QPushButton:hover:!pressed
{
	background-color: #e60000;
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/files/images/x_close.png</normaloff>:/files/images/x_close.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>20</width>
      <height>20</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="btnMinimize">
    <property name="geometry">
     <rect>
      <x>782</x>
      <y>2</y>
      <width>40</width>
      <height>40</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton
{
	border-radius: 0px;
	border: none;
	background-color: #6e6e6e;
}
QPushButton:pressed
{
	background-color: #6e6e6e;
}
QPushButton:hover:!pressed
{
	background-color: #5e5e5e;
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="resources.qrc">
      <normaloff>:/files/images/-_minimum.png</normaloff>:/files/images/-_minimum.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>25</width>
      <height>25</height>
     </size>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>DraggableWidget</class>
   <extends>QWidget</extends>
   <header location="global">draggablewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
